lib/modules/manager/npm/extract/pnpm.ts

```diff
@@ -135,3 +142,124 @@ export async function detectPnpmWorkspaces(
     }
   }
 }
+
+export async function getPnpmLock(filePath: string): Promise<LockFile> {
+  // TODO #7154
+  const pnpmLockRaw = (await readLocalFile(filePath, 'utf8'))!;
+  try {
+    const lockParsed = load(pnpmLockRaw);
+    if (!isPnpmLockfile(lockParsed)) {
+      throw new Error('Invalid or empty lockfile');
+    }
+    logger.trace({ lockParsed }, 'pnpm lockfile parsed');
+
+    // field lockfileVersion is type string in lockfileVersion = 6 and type number in < 6
+    const lockfileVersion: number = is.number(lockParsed.lockfileVersion)
+      ? lockParsed.lockfileVersion
+      : parseFloat(lockParsed.lockfileVersion);
+
+    const lockedVersions: Record<string, string> = {};
+    const packagePathRegex = regEx(
+      /^\/(?<packageName>.+)(?:@|\/)(?<version>[^/@]+)$/
+    ); // eg. "/<packageName>(@|/)<version>"
+
+    for (const packagePath of Object.keys(lockParsed.packages ?? {})) {
+      const result = packagePath.match(packagePathRegex);
+      if (!result?.groups) {
+        logger.debug(`Invalid package path ${packagePath}`);
```

Comment:
This could be noisy, is it likely to happen?

Reply:
No